engine/src/main/java/org/terasology/audio/AudioManager.java

```diff
@@ -123,8 +133,12 @@
      * @param position The new position
      * @param orientation The new orientation (in a quaternion)
      * @param velocity The new velocity
+     * @deprecated This method is scheduled for removal in an upcoming version.
+     *             Use the JOML implementation instead: {@link #updateListener(Vector3fc, Quaternionf, Vector3fc)}.
      */
+    @Deprecated
     void updateListener(Vector3f position, Quat4f orientation, Vector3f velocity);
+    void updateListener(Vector3fc position, Quaternionf orientation, Vector3fc velocity);
```

Comment:
can we change it to use Quaternionfc. I don't think we modify the orientation in anyway so the interface would make more sense here.

Reply:
yes, of course.